examples/sparse/gat.py

```diff
@@ -0,0 +1,134 @@
+"""
+[Graph Attention Networks]
+(https://arxiv.org/abs/1710.10903)
+"""
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from dgl.data import CoraGraphDataset
+from dgl.mock_sparse import bspmm, create_from_coo, identity
+from torch.optim import Adam
+
+
+class GATConv(nn.Module):
+    def __init__(self, in_size, out_size, num_heads, dropout):
+        super().__init__()
+
+        self.out_size = out_size
+        self.num_heads = num_heads
+
+        self.dropout = nn.Dropout(dropout)
+        self.W = nn.Linear(in_size, out_size * num_heads)
+        self.a_l = nn.Parameter(torch.zeros(1, out_size, num_heads))
+        self.a_r = nn.Parameter(torch.zeros(1, out_size, num_heads))
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        gain = nn.init.calculate_gain("relu")
+        nn.init.xavier_normal_(self.W.weight, gain=gain)
+        nn.init.xavier_normal_(self.a_l, gain=gain)
+        nn.init.xavier_normal_(self.a_r, gain=gain)
+
+    def forward(self, A_hat, Z):
+        Z = self.dropout(Z)
+        Z = self.W(Z).view(Z.shape[0], self.out_size, self.num_heads)
+        # a^T [Wh_i || Wh_j] = a_l Wh_i + a_r Wh_j
```

Comment:
nit: which lines implement this formula? L37-L39? Add blank lines before and after the code block.